### PR TITLE
chore: add cx bulk query tool, fix bulk patient insert tool

### PR DESCRIPTION
**Pull request by Goncharo**
refs. metriport/metriport-internal#1040

### Dependencies

n/a

### Description

add cx bulk query tool, fix bulk patient insert tool

### Release Plan

asap, nothing special

### CODE REVIEW

File: packages/utils/src/bulk-query-patients.ts
Comment (leite08) [inline packages/utils/src/bulk-query-patients.ts:1]: Add a comment at the top to explain what this does?

File: packages/utils/src/bulk-query-patients.ts [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+    await sleep(200 + Math.random() * patientChunkDelayJitterMs);
+    const queryStartTime = Date.now();
+    while (Date.now() - queryStartTime < maxQueryDurationMs) {
+      const docQueryStatus = await metriportAPI.getDocumentQueryStatus(patient.id);
+      // ensure at least 1 doc was downloaded, and both the download + conversion have completed
+      if (
+        docQueryStatus.download &&
+        docQueryStatus.download.status === "completed" &&
+        docQueryStatus.download.successful &&
+        docQueryStatus.download.successful > 0 &&
```
Comment (leite08) [inline packages/utils/src/bulk-query-patients.ts:67]: Why looking at successful, wouldn't `status !== processing` suffice?

File: packages/utils/src/bulk-query-patients.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  fs.appendFileSync(
+    csvName,
+    `${patient.id},${patient.firstName},${
+      patient.lastName
+    },${state},${docQueryAttempts},${docCount},${fhirResourceCounts},${replaceAll(
+      JSON.stringify(fhirResourceCount),
+      ",",
+      " "
+    )},${status}\n`
+  );
```
Comment (leite08) [inline packages/utils/src/bulk-query-patients.ts:101]: NABD, but ideally would prep these into individual variables so we're not trying to understand how each gets build/purpose, as well as how they get arranged

File: packages/utils/src/bulk-query-patients.ts [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    const docQueries: Promise<void>[] = [];
+    for (const patient of chunk) {
+      docQueries.push(queryDocsForPatient(patient));
+    }
+    await Promise.allSettled(docQueries);
+
+    console.log(`>>> Sleeping for ${delayTime} ms before the next chunk...`);
+    await new Promise(f => setTimeout(f, delayTime));
+  }
+}
```
Comment (leite08) [inline packages/utils/src/bulk-query-patients.ts:122]: nit: log saying its done and maybe some result/count if useful

File: packages/utils/src/bulk-query-patients.ts (hunk 1 above)
Comment (Goncharo) [reply packages/utils/src/bulk-query-patients.ts:67]: This is purposefully here to trigger a retry if we initially get 0 docs - due to the behavior we've previously seen

File: packages/utils/src/bulk-query-patients.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  const state = Array.isArray(patient.address) ? patient.address[0].state : patient.address.state;
+  fs.appendFileSync(
+    csvName,
+    `${patient.id},${patient.firstName},${
+      patient.lastName
+    },${state},${docQueryAttempts},${docCount},${fhirResourceCounts},${replaceAll(
+      JSON.stringify(fhirResourceCount),
+      ",",
+      " "
+    )},${status}\n`
```
Comment (leite08) [inline packages/utils/src/bulk-query-patients.ts:100]: not clear why need both `fhirResourceCounts` and `fhirResourceCount` on the output

also, these names are easy to get confused, super similar and not just a list and an individual entry :)

Comment (Goncharo) [reply packages/utils/src/bulk-query-patients.ts]: Fixed names, we need both b/c one is a total while the other is counts of each individual resource

**Pull request merged**